pkg/didcomm/protocol/introduce/states.go

```diff
@@ -29,6 +33,14 @@ const (
 	stateNameWaiting  = "waiting"
 )
 
+type provider interface {
+	dispatcher.Outbound
+}
+
+type providerImpl struct {
```

Comment:
All the dependencies should come from aries.Context(). We shouldn't implement these in protocols.

Reply:
(more specifically, the protocol should define an interface of the subset of Context that it needs).

Reply:
ok, I will do that more clear :)

Reply:
Done!